### ISSUE TITLE
Default generic value T{} is not recognized in if condition

### ISSUE THREAD

**Issue by avitkauskas** [vlang/v#22677]
### Describe the bug

in `if e != T{} {...}` T{} is formatted wrongly by the formatter and does not complile.

### Reproduction Steps

```v
struct St[T] {
mut:
	a T
}

fn (mut s St[T]) f(e T) {
        // notice the wrong formatting of `if e != T{}`
	if e != T{
        }
	{
		s.a = e
	}
}

fn main() {
	mut s := St[int]{}
	s.f(1)
}
```

### Expected Behavior

It should be left formatted as `if e != T{}` and it should format.

### Current Behavior

bug2.v:7:10: error: undefined ident: `T`
    5 |
    6 | fn (mut s St[T]) f(e T) {
    7 |     if e != T {
      |             ^
    8 |     }
    9 |     {
bug2.v:7:5: error: non-bool type `void` used as if condition
    5 |
    6 | fn (mut s St[T]) f(e T) {
    7 |     if e != T {
      |        ~~~~~~
    8 |     }
    9 |     {

### V version

V 0.4.8 8ffcc34

### Environment details (OS name and version, etc.)

macOS
> [!NOTE]
> You can use the 👍 reaction to increase the issue's priority for developers.
>
> Please note that only the 👍 reaction to the issue itself counts as a vote.
> Other reactions and those to comments will not be taken into account.

**Comment by avitkauskas** [comment]
I confirm this is fixed by #22682.

**Comment by felipensp** [comment]
Thanks for testing @avitkauskas
